charts/karpenter-crd/templates/karpenter.sh_nodeclaims.yaml

```diff
@@ -380,440 +389,6 @@ spec:
       storage: true
       subresources:
         status: {}
-    - additionalPrinterColumns:
-        - jsonPath: .metadata.labels.node\.kubernetes\.io/instance-type
-          name: Type
-          type: string
-        - jsonPath: .metadata.labels.topology\.kubernetes\.io/zone
-          name: Zone
-          type: string
-        - jsonPath: .status.nodeName
-          name: Node
-          type: string
-        - jsonPath: .status.conditions[?(@.type=="Ready")].status
-          name: Ready
-          type: string
-        - jsonPath: .metadata.creationTimestamp
-          name: Age
-          type: date
-        - jsonPath: .metadata.labels.karpenter\.sh/capacity-type
-          name: Capacity
-          priority: 1
-          type: string
-        - jsonPath: .metadata.labels.karpenter\.sh/nodepool
-          name: NodePool
-          priority: 1
-          type: string
-        - jsonPath: .spec.nodeClassRef.name
-          name: NodeClass
-          priority: 1
-          type: string
-      name: v1beta1
```

Comment:
Apologies if I asked this already last year and it was answered:

Users that had originally created Pools/Claims with v1beta1 will have that value in `.status.storedVersions`. Applying new CRD version without `v1beta1` will be rejected by kube-apiserver unless `.status.storedVersions` has `v1beta1` removed from it.

AFAIK, it's safe to remove `v1beta` from `.status.storedVersions` now, because it's not the storage version anymore - but something has to actually _do_ that.

What does it?

Reply:
Ah, this does it (from internal chat): https://github.com/kubernetes-sigs/karpenter/pull/1689

from Karpenter 1.0.6 - so we need to update to that first.

Reply:
Migration controller https://github.com/kubernetes-sigs/karpenter/pull/1689 (1.0.6+)

Reply:
Chatted offline. Appears we need to address this by doing a rollout of `v1.0.6` first. After that point, this should be addressed, and we can do this `v1.1.x` change.

Reply:
Reactivating this just to make sure that I remember correctly, we actually determined that the migration controller was enabled since `v1.0.5`, _and_ the NAP-based identity had the correct permissions for it, so at least _in theory_ NAP should be safe to remove the v1beta1 CRDs + webhooks.

Have we tested that though and confirmed?

What about for selfhosted users installing via Helm? We only enabled the correct permissions for the operator identity in https://github.com/Azure/karpenter-provider-azure/pull/755, which I don't think has been released yet?

Do we need to make sure that's released first? What's the user-upgrade story we ask users to go through for selfhosted? Do we expect they update 1 version at a time, or can they jump versions?

Reply:
> migration controller was enabled since v1.0.5

Since 1.0.6, would have to be released first

> Have we tested that though and confirmed?

Not yet, planned

> What about for selfhosted users installing via Helm?

Yes, need to release a version with migration controller first, a version with webhooks & migration controller should not/cannot be skipped, and need to have release notes similar to https://karpenter.sh/v1.0/upgrading/v1-migration/#before-upgrading-to-v110

Reply:
There's been a new selfhosted release, so resolving this.